packages/@storefront/core/test/unit/services/tracker.ts

```diff
@@ -190,12 +190,49 @@ suite('Tracker Service', ({ expect, spy, stub, itShouldExtendBaseService }) => {
 
       expect(override).to.be.calledWithExactly(value, currentEvent);
     });
+
+    it('should call attachGbiEventMetadata() with override', () => {
+      const event: any = { a: 'b' };
+      const currentEvent = { c: 'd' };
+      const result = { e: 'f', metadata: [{ i: 'j' }] };
+      const override = spy(() => result);
+      const value = { g: 'h' };
+      const attachGbiEventMetadataSpy = spy();
+      service.attachGbiEventMetadata = attachGbiEventMetadataSpy;
+      stub(service, 'addMetadata').withArgs(event).returns(currentEvent);
+
+      service.buildEvent(override, event, value);
+
+      expect(attachGbiEventMetadataSpy).to.be.calledWith(result);
+    });
+  });
+
+  describe('attachGbiEventMetadata()', () => {
+    const GBI_EVENT = { key: 'gbi', value: 'true' };
+
+    it('should add the GBI_EVENT object to the override event metadata', () => {
+      const override = { e: 'f', metadata: [{ i: 'j' }] };
+
+      const overriddenEvent = service.attachGbiEventMetadata(<any>override);
+
+      expect(overriddenEvent.metadata).to.eql([GBI_EVENT, ...override.metadata]);
+    });
+
+    it('should not add the GBI_EVENT if it is already present in the event metadata', () => {
+      const override = { e: 'f', metadata: [GBI_EVENT, { i: 'j' }] };
+
+      const overriddenEvent = service.attachGbiEventMetadata(<any>override);
+
+      expect(overriddenEvent.metadata).to.be.eql([...override.metadata]);
+    });
   });
 
   describe('sendSearchEvent()', () => {
     const id = '12345';
     const metadata = [{ a: 'b' }];
 
+    beforeEach(() => service.attachGbiEventMetadata = <T>(override: T): T => override);
```

Comment:
Instead of repeating this for each `Event` function, perhaps it would be better to group the `Event` tests into a `describe` block and have this done once?

Reply:
Done.